gradoop-flink/src/main/java/org/gradoop/flink/algorithms/gelly/randomjump/KRandomJumpGellyVCI.java

```diff
@@ -0,0 +1,217 @@
+   * Probability for jumping to a random vertex instead of walking to a random neighbor.
+   */
+  private final double jumpProbability;
+
+  /**
+   * Relative amount of vertices to visit at least.
+   */
+  private final double percentageToVisit;
+
+  /**
+   * DataSet holding the mapping for a long index to its vertex gradoop id.
+   */
+  private DataSet<Tuple2<Long, GradoopId>> indexToVertexIdMap;
+
+  /**
+   * DataSet holding the mapping for a long index to its edge gradoop id.
+   */
+  private DataSet<Tuple2<Long, GradoopId>> indexToEdgeIdMap;
+
+  /**
+   * Creates an instance of KRandomJumpGellyVCI.
+   *
+   * @param k Number of starting vertices.
+   * @param maxIterations Value for maximum number of iterations for the algorithm.
+   * @param jumpProbability Probability for jumping to random vertex instead of walking to random
+   *                       neighbor.
+   * @param percentageToVisit Relative amount of vertices to visit at least.
+   */
+  public KRandomJumpGellyVCI(int k, int maxIterations, double jumpProbability,
+    double percentageToVisit) {
+    checkArgument(k >= 1,
+      "at least 1 starting vertex is needed, k must be equal or greater 1");
+    checkArgument(maxIterations > 0,
+      "maxIterations must be greater than 0");
+    checkArgument(jumpProbability >= 0d && jumpProbability <= 1d,
+      "jumpProbability must be equal/greater than 0.0 and smaller/equal 1.0");
+    checkArgument(percentageToVisit > 0d && percentageToVisit <= 1d,
+      "percentageToVisit must be greater than 0.0 and smaller/equal 1.0");
+    this.k = k;
+    this.maxIterations = maxIterations;
+    this.jumpProbability = jumpProbability;
+    this.percentageToVisit = percentageToVisit;
+  }
+
+  @Override
+  public Graph<Long, VCIVertexValue, Long> transformToGelly(LogicalGraph graph) {
+    this.currentGraph = graph;
+
+    indexToVertexIdMap = DataSetUtils.zipWithIndex(graph.getVertices().map(new Id<>()));
+    indexToEdgeIdMap = DataSetUtils.zipWithIndex(graph.getEdges().map(new Id<>()));
+
+    DataSet<Vertex<Long, VCIVertexValue>> vertices = indexToVertexIdMap
+      .map(new LongIdToGellyVertexWithVCIValueMap());
+
+    DataSet<Edge<Long, Long>> edges = graph.getEdges()
+      .join(indexToVertexIdMap)
+      .where(new SourceId<>()).equalTo(1)
+      .with(new LongIdWithEdgeToTupleJoin())
+      .join(indexToVertexIdMap)
+      .where(1).equalTo(1)
+      .with(new ReplaceTargetWithLongIdJoin())
+      .join(indexToEdgeIdMap)
+      .where(2).equalTo(1)
+      .with(new LongIdTupleToGellyEdgeWithLongValueJoin());
+
+    return Graph.fromDataSet(vertices, edges, graph.getConfig().getExecutionEnvironment());
+  }
+
+  @Override
+  public LogicalGraph executeInGelly(Graph<Long, VCIVertexValue, Long> gellyGraph)
+    throws Exception {
+
+    Set<Long> randomStartIndices = new HashSet<>();
+    while (randomStartIndices.size() < k) {
+      long randomLongInBounds = (long) (Math.random() * (gellyGraph.numberOfVertices() - 1L));
```

Comment:
Initialize a param with result of `gellyGraph.numberOfVerteices()` instead of calling the method multiple times. Keep in mind that this method presents a `sink` in the flink dataflow.